src/http/client.rs

```diff
@@ -4381,7 +4381,7 @@ impl Http {
     pub async fn request(&self, req: Request<'_>) -> Result<ReqwestResponse> {
         let method = req.method.reqwest_method();
         let response = if let Some(ratelimiter) = &self.ratelimiter {
-            ratelimiter.perform(req).await?
+            Box::pin(ratelimiter.perform(&req)).await?
```

Comment:
I think boxing large futures does nothing to actually address the size problem. The data is just now on the heap and incurs an allocation.

Reply:
This shrinks size of the future for the branch where a rate limiter is disabled, which is required for large bots using nirn/another proxy.

Reply:
I was curious what this would actually save. With the changes in the PR, this should really only reduce the size of the `Http::request` future by about 200 bytes -- while allocating almost 2kb for the ratelimiter case. That doesn't seem worth it.

Partial output with `Zprint-type-sizes`:
```
type: `{async fn body of http::client::Http::request()}`: 2264 bytes, alignment: 8 bytes
    variant `Suspend0`: 711 bytes // boxed ratelimiter await
    variant `Suspend1`: 2263 bytes // second largest await
type: `{async fn body of http::ratelimiting::Ratelimiter::perform()}`: 1744 bytes, alignment: 8 bytes
```

That said, why not put the ratelimiter behind the feature gate?

Reply:
My impression with futures & async is that there are no sweeping giant single gains to be made. That the problems just come from a zillion smaller bits adding on top of eachother and you have to look at that and see what systematic changes can be made in many places, little by little.

Reply:
That said, from my POV, this does not seem like the right solution. Rate limiting is very common, and we're going to pay a bunch of cost in the common case for a lot of people as a result. What do you instead think about doing some strategic cold-path future outlining/boxing inside of the rate limiter code? That definently has hot and cold paths.